### PR TITLE
chore: initial k8s rocm support [CM-367]

### PR DESCRIPTION
**Pull request by NicholasBlaskey**
## Description

Adds initial experimental support for amd gpus on k8s.

## Test Plan

Unit tests cover this change

Also manually verified on amd hardware. Automated e2e tests were deemed not in scope due to hardware availability issues

blocked on https://github.com/determined-ai/environments/pull/275. We just need the environment images to land in the same release as this. Some additional docs will be added in the environments pr detailing what our rocm environment does and does not support. 

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/reference/deploy/master-config-reference.rst [hunk 1]
```diff
@@ -403,14 +403,21 @@ resource pool ``max_slots_per_pod``.
 ``slot_type``
 -------------
 
-Resource type used for compute tasks. Defaults to ``cuda``.
+Resource type used for compute tasks. Valid options are ``gpu``, ``cuda``, ``cpu``, or ``rocm``.
+Defaults to ``cuda``.
```
Comment (kkunapuli) [inline docs/reference/deploy/master-config-reference.rst:407]: Does it default to `cuda` or `gpu`? The helm config reference says `gpu`?

File: docs/release-notes/rocm-k8s-gpu.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+:orphan:
+
+**New Features**
+
+-  Kubernetes: Experimental support for AMD ROCM GPUs is now available for Kubernetes. To use set
+   ``slotType=rocm``. See :ref:`helm-config-reference` for more details.
```
Comment (kkunapuli) [inline docs/release-notes/rocm-k8s-gpu.rst:6]: nit: I think the docs style guideline prefers `visit` over `see`.

File: master/internal/config/resource_manager_config.go [hunk 3]
```diff
@@ -303,11 +303,9 @@ func (k *KubernetesResourceManagerConfig) UnmarshalJSON(data []byte) error {
 func (k KubernetesResourceManagerConfig) Validate() []error {
 	var checkSlotType error
 	switch k.SlotType {
-	case device.CPU, device.CUDA:
-	case device.ROCM:
-		checkSlotType = errors.Errorf("rocm slot_type is not supported yet on k8s")
+	case device.CPU, device.CUDA, device.ROCM:
```
Comment (kkunapuli) [inline master/internal/config/resource_manager_config.go:306]: Does the code detect when the user inputs `gpu` and switches it to `cuda`? This makes it seem like `gpu` isn't actually accepted for `slotType`?

**Review by kkunapuli — APPROVED** [review]
LGTM! I left a couple questions, but nothing blocking.

File: docs/reference/deploy/master-config-reference.rst (hunk 1 above)
Comment (NicholasBlaskey) [reply docs/reference/deploy/master-config-reference.rst:407]: The master config defaults to ```cuda```, helm defaults to ```gpu```

File: master/internal/config/resource_manager_config.go (hunk 3 above)
Comment (NicholasBlaskey) [reply master/internal/config/resource_manager_config.go:306]: Yeah we switch it here

https://github.com/determined-ai/determined/blob/80822ebe40e3516ca987872429bb6cac97de7200/master/internal/config/resource_manager_config.go#L297

Comment (kkunapuli) [reply master/internal/config/resource_manager_config.go:306]: Thanks!!

**Pull request merged**